database/sqltypes/sqltypes.go

```diff
@@ -577,11 +577,11 @@ var encodeRef = map[byte]byte{
 }
 
 func init() {
-	for i, _ := range SqlEncodeMap {
```

Comment:
theses are intentional.  the optional second value is pretty evil and has caused us pain in the past (when both key/index and value are the same type)

Reply:
I understand your point, will update accordingly if rest is fine.